### PR TITLE
feat(php-checks): add generic php workflow

### PR DESCRIPTION
**Pull request by jpeters8889**
I've created this generic PHP Workflow file that can be used across the business. I created and tested this in the Public API before copying the file to here for this PR.

The Workflow will install all composer dependencies, then check if a `php-cs-fixer.dist.php` file exists, if it does, it will run PHP CS Fixer.

It then checks for a `phpstan.neon` file, and runs PHPStan if present

It then finally checks for `phpunit.xml` file, and runs PHPUnit if present.

The three checks can also be skipped using an input from the child workflow using the PHP Workflow, by passing in `skip-cs-fixer`, `skip-phpstan` or `skip-phpunit` flags, the idea being that when first setting up a project some of the workflows may fail, for example PHPStan, but we don't want that to cause the workflow to fail just yet, but once PHPStan is working locally and any code changes made to get it working, then we'll stop skipping PHPStan and enforce that rule.

### CODE REVIEW

**Review by kieranbrown — CHANGES_REQUESTED** [review]
Love this dude! Got some minor feedback.

- We typically use scopes in the commit messages for the workflows, so the commit message would be `feat(php-workflow): initial implementation` or something.
  - The Jira ticket is inferred from the branch so not required in the commit message.
  - You'd need to add this to the available scopes in commitlintrc.
- Could we rename to `php-checks` to keep inline with `tlz-checks`? php-workflow is a bit generic as it could be literally anything really. Inline with that the commit message would be `feat(php-checks): ...`.
- Could we move each tool into it's own job?
  - It's safer to do this as if a step completely fails it will stop processing.
  - It also allows the jobs to run in parallel.
- It's worth adding some concurrency options onto here. I'd also potentially use `cancel-in-progress` as if a new commit is pushed it's worthless running checks on old code. See example here: https://github.com/clickdealer/github-actions/blob/v1/.github/workflows/sonar-scanner.yaml#L21-L23 however I'd be inclined to add this to the root level so it's inherited from all of the jobs.
- I would hard-code the `runs-on` to be `ubuntu-latest` as we're decommissioning our self-hosted runners soon.
- The inputs that enable certain steps (jobs after feedback) would be better as a positive and not a negative, so you'd set `phpunit: false` or `enable-phpunit: false` to disable it.
- I like how we're dynamically running the tools based on config files but what if the config files don't exist? Won't PHPStan run without a config file and just use defaults?

File: .github/workflows/php-checks.yml [hunk 1]
```diff
@@ -0,0 +1,108 @@
+name: php-workflow
```
Comment (kieranbrown) [inline .github/workflows/php-checks.yml:1]:
```suggestion
name: php-checks
```

File: .github/workflows/php-checks.yml [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        uses: andstor/file-existence-action@v1
+        id: phpstan_config_file
+        with:
+          files: phpstan.neon
+
+      - name: Run PHPStan
+        if: inputs.enable-phpstan == true && steps.phpstan_config_file.outputs.files_exists == 'true'
+        run: ./vendor/bin/phpstan analyse
+
+  run-phpunit:
```
Comment (kieranbrown) [inline .github/workflows/php-checks.yml:84]:
```suggestion
  phpunit:
```

File: .github/workflows/php-checks.yml [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+        run: ./vendor/bin/php-cs-fixer fix --dry-run --config=php-cs-fixer.dist.php --allow-risky=yes --stop-on-violation
+
+  run-phpstan:
+    runs-on: ubuntu-latest
+    if: inputs.enable-phpstan == true
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.head_ref }}
```
Comment (kieranbrown) [inline .github/workflows/php-checks.yml:65]:
```suggestion
      - name: Checkout
        uses: actions/checkout@v3
```

File: .github/workflows/php-checks.yml [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+        run: ./vendor/bin/phpstan analyse
+
+  run-phpunit:
+    runs-on: ubuntu-latest
+    if: inputs.enable-phpunit == true
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.head_ref }}
```
Comment (kieranbrown) [inline .github/workflows/php-checks.yml:91]:
```suggestion
      - name: Checkout
        uses: actions/checkout@v3
```

**Review by kieranbrown — CHANGES_REQUESTED** [review]
looking really good, I've added some suggestions for some minor touch ups

it's worth having a think about how we could annotate the code and provide job summaries, this is usually a big benefit of bringing in an action that somebody else has already written as it could include annotations/job summary etc.

Here's a blog post on job summaries:
https://github.blog/2022-05-09-supercharging-github-actions-with-job-summaries/ 

Also here's an example of annotations:
image

**Review by kieranbrown — APPROVED** [review]
lgtm - will leave it with you in regards to how far you want to take this, as discussed on Teams it would be great to get job summary / code annotations in here too.

I think job summaries is a bigger task however getting annotations in there for cs-fixer/phpstan shouldn't be too tricky.

For phpstan you can set the output format to GitHub https://phpstan.org/user-guide/output-format.

For php-cs-fixer I found how nextcloud achieved this - https://github.com/nextcloud/server/pull/21813

**Comment by jpeters8889** [comment]
I've put the above comments on a card for a future sprint, gonna treat this as the first version of the workflow which works, and then in a future sprint build on it with the ideas above.

**Pull request merged**